Magento2/Sniffs/Legacy/ModuleXMLSniff.php

```diff
@@ -0,0 +1,91 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento2\Sniffs\Legacy;
+
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Sniffs\Sniff;
+use SimpleXMLElement;
+
+/**
+ * Test for obsolete nodes/attributes in the module.xml
+ */
+class ModuleXMLSniff implements Sniff
+{
+    /**
+     * Error violation code.
+     *
+     * @var string
+     */
+    protected $warningCode = 'FoundObsoleteAttribute';
+    
+    /**
+     * @inheritdoc
+     */
+    public function register(): array
+    {
+        return [
+            T_INLINE_HTML
+        ];
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function process(File $phpcsFile, $stackPtr)
+    {
+        $line = $phpcsFile->getTokens()[$stackPtr]['content'];
+        if (strpos(trim($line), '<module ') === false) {
```

Comment:
It would be good to validate the file path as well

Reply:
That's already done in the ruleset